src/shim.c

```diff
@@ -0,0 +1,42 @@
+/* shim.c
+   Rémi Attab (remi.attab@gmail.com), 14 Sep 2017
+   FreeBSD-style copyright and disclaimer apply
+*/
+
+#include "rill.h"
+
+#include <assert.h>
+
+
+// -----------------------------------------------------------------------------
+// error
+// -----------------------------------------------------------------------------
+// Thread locals are a nightly feature so shim it for now.
```

Comment:
What do you mean by that?
[`thread_local`](https://doc.rust-lang.org/stable/std/macro.thread_local.html) is stable, no?

Reply:
So that thread_local is a macro and doesn't really allow you to ffi with a C thread-local. If I remember correctly nightly add an annotation for thread local linking or whatever. I did try a couple of things but it quickly got annoying and decided to revert to a shim to avoid headaches.